src/atom_finder/classifier/operator-precedence.clj

```diff
@@ -0,0 +1,131 @@
+                IASTBinaryExpression/op_lessThan :compare
+                IASTBinaryExpression/op_lessEqual :compare
+                IASTBinaryExpression/op_equals :compare
+                IASTBinaryExpression/op_notequals :compare
+                IASTBinaryExpression/op_binaryAnd :bitwise
+                IASTBinaryExpression/op_binaryXor :bitwise
+                IASTBinaryExpression/op_binaryOr :bitwise
+                IASTBinaryExpression/op_logicalAnd :and
+                IASTBinaryExpression/op_logicalOr :or}]
+           (->> node .getOperator operator-group-table))))
+
+
+(defn specific-confusing-operator-combination?
+  "if the combination of groups of operators exists in this set, then the combination is confusing"
+  [combination]
+  (some #(= % combination) (map sort #{[:de_incr :pointer] [:multiply :add] [:arith_unary :add] [:arith_unary :multiply] [:and :add] [:and :multiply] [:and :arith_unary] [:or :add] [:or :multiply] [:or :arith_unary] [:or :and] [:not :add] [:not :multiply] [:not :arith_unary] [:not :and] [:not :or]
+;;[:compare :and] [:compare :or] // Underclassify
+[:compare :not] [:compare :compare] [:pointer :add] [:cond :arith_unary] [:cond :and] [:cond :or] [:cond :not] [:cond :compare] [:cond :cond] [:non-asso :add] [:non-asso :multiply] [:non-asso :arith_unary] [:non-asso :and] [:non-asso :or] [:non-asso :not] [:non-asso :non-asso] [:field_ref :pointer]})))
+
+(def always-confusing-operator-groups
+  "If any of these operator groups is used along with another operator, then it is confusing"
+  #{:bitwise :comma})
+
+(defn confusing-operator-combination?
+  "Is this combination of the operator groups confusing?"
+  [combination]
+  (or (specific-confusing-operator-combination? combination)
+      (some (partial contains? always-confusing-operator-groups) combination)))
+
+(defn operator-group-pair?
+  [node]
+  (and (operator-group node)
+       (some operator-group (children node))))
+
+(defn group-pair
+  "returns a collection of operator group pairs between the node and its children
+  , if a second parameter is passed, use that as the collection instead of the children"
+  ([node]
+   (->> node
+        children
+        (map operator-group)
+        (remove nil?)
+        (map (fn [child-group] (sort [(operator-group node) child-group])))))
+
+  ([node collection]
+   (->> collection
+        (map operator-group)
+        (remove nil?)
+        (map (fn [child-group] (sort [(operator-group node) child-group]))))))
+
+
+;;
+;;The following 3 functions are for the binary operator special case
+;;
+(defn rvalue-unary-operator?
+  "Is this a unary operator that can operate on an rvalue? (! - + *)"
+  [node]
+  (let [node-group (operator-group node)]
+    (or (= :arith_unary node-group)
+        (= :not node-group)
+        (= :pointer node-group))))
+
+(defn unbracktted-unary-in-children
+  "Is this node not a bracket and constains unary operator in its children?"
+  [node]
+  (if (not= "()" (write-node node))
```

Comment:
I'd love to see a predicate `parens?`, but I'd probably implement it `(instance? IASTParenthesisExpression)` or whatever.